### PR TITLE
DUI3-326:  Add civil3d 2024 converter

### PR DESCRIPTION
**Pull request by clairekuang**
Adds the Civil3d 2024 converter.

This pr includes **required changes to container registration** in order to support converter injection into a connector vertical.
These changes should be reflected in the other DUI3 converter DI projects before this pr is merged.

First pipe sent to speckle from Civil3d: https://app.speckle.systems/projects/b53a53697a/models/719612f9da

## Changes:

- sdk container registration
- missing autocad conversions
- civil3d 2024 converter 

## To-do before merge:

- [x] Fix container registration for all other DUI3 converter DI projects

### CODE REVIEW

**Review by oguzhankoral — CHANGES_REQUESTED** [review]
Only reason requesting change on static class, but not sure it was a hack for multiple converter registration?
Q: why we removed ITypedConversions and extracted them? In order to use them on Civil3D too?

File: DUI3-DX/Converters/Autocad/Speckle.Converters.Autocad2024.DependencyInjection/AutocadConverterModule.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 using Speckle.Converters.Common;
 using Speckle.Converters.Common.DependencyInjection;
 
-namespace Speckle.Converters.Autocad20243.DependencyInjection;
+namespace Speckle.Converters.Autocad2024.DependencyInjection;
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Converters/Autocad/Speckle.Converters.Autocad2024.DependencyInjection/AutocadConverterModule.cs:8]: I think it's a tester for when we do.

File: DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToSpeckle/Raw/DBCurveToSpeckleRawConverter.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Core.Models;
+
+namespace Speckle.Converters.Autocad.ToSpeckle.Raw;
+
+public class DBCurveToSpeckleRawConverter : ITypedConverter<ADB.Curve, Objects.ICurve>, ITypedConverter<ADB.Curve, Base>
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToSpeckle/Raw/DBCurveToSpeckleRawConverter.cs:7]: shouldn't be any different than just 1

File: DUI3-DX/Converters/Autocad/Speckle.Converters.Autocad2024.DependencyInjection/AutocadConverterModule.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (clairekuang) [reply DUI3-DX/Converters/Autocad/Speckle.Converters.Autocad2024.DependencyInjection/AutocadConverterModule.cs:8]: Yes the autocad 2024 converter is required for civil3d 2024

**Comment by clairekuang** [comment]
> Q: why we removed ITypedConversions and extracted them? In order to use them on Civil3D too?

There is no difference between having a converter implement both `ITopLevelConverter` and `ITypedConverter`, or to separate out the typed converter class to be called from the top level converter. I thought the second option is cleaner, so we can see at a glance all our typed converters.

**Review by oguzhankoral — APPROVED** [review]

**Pull request merged**